text/0052-resource-importing-support.md

```diff
@@ -0,0 +1,272 @@
+
+### Why should I use this feature?
+
+When an already existing AWS resource needs to be brought under CDK management
+the most convenient way of achieving this is using this feature (`cdk import`).
+
+## Internal FAQ
+
+### Why are we doing this?
+
+This solution is exposing an existing [CloudFormation resource
+import](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-import.html)
+feature in a CDK native way. CDK's goal is to provide convenient abstraction
+over sometimes complex CloudFormation aspects and this new feature is aligned
+with this goal. Additionally, other IaC orchestrators' also provide a
+corresponding feature, for example:
+
+  * [terraform import](https://www.terraform.io/cli/import)
+  * [pulumi import](https://www.pulumi.com/docs/guides/adopting/import/#pulumi-import-command)
+
+### Why should we _not_ do this?
+
+The feature might be considered independent enough to provide a separate tool
+for it (e.g. `cdk-import` node package).
+
+### What is the technical solution (design) of this feature?
+
+  * [Proof-of-Concept PR](https://github.com/aws/aws-cdk/pull/17666)
+
+The solution leverages CloudFormation resource import API. The API is very
+similar to the regular deploy API, consisting of ChangeSet creation and
+ChangeSet execution actions.
+
+  * ChangeSet of type `IMPORT` is created (for deploy it is `CREATE` or 
+    `UPDATE`)
+  * on ChangeSet creation, an additional structure has to be passed with the
+    action, an array of [ResourceToImport](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_ResourceToImport.html)
+    structures to instruct CloudFormation what actual existing resources to
+    import as corresponding logical resources in the template
+  * ChangeSet is executed as on regular deployment
+
+The solution is leveraging existing wrappers around ChangeSet creation and
+execution in `aws-cdk` codebase, extending them with the options required for
+importing.
+
+Following characteristics/limitations of the API pose a challenge to the
+solution:
+
+#### Inconsistent resource identifiers between resource types
+
+In the ResourceToImport structure, each type of resource is identified by a set
+of resource properties (typically of length 1 - by a single property). The
+property name is different between different resource types (e.g. `BucketName`
+for S3 buckets, or `VpcId` for VPCs).
+
+To be able to put the structure together, the import logic must be able to
+identify the required properties and then look up the values. For some
+resources, this can be automatically looked up from the CDK construct
+configuration (e.g. an S3 bucket having specific `bucketName` configured) but
+for some resources, the code has to rely on user input.
+
+To address this, the solution is divided into two steps:
+
+##### Physical resource identification
+
+First, identify what properties must be looked up/asked for for each resource to
```

Comment:
Sweet! Does this mean we don't need to hardcode support for each resource type anymore?

Reply:
Exactly, this is a very neat feature from CFN :)